src/System.Net.Http.Formatting/Internal/ByteRangeStream.cs

```diff
@@ -132,33 +182,49 @@ public override void EndWrite(IAsyncResult asyncResult)
             throw Error.NotSupported(Properties.Resources.ByteRangeStreamReadOnly);
         }
 
+        public override Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+        {
+            throw Error.NotSupported(Properties.Resources.ByteRangeStreamReadOnly);
+        }
+
         public override void WriteByte(byte value)
         {
             throw Error.NotSupported(Properties.Resources.ByteRangeStreamReadOnly);
         }
 
         /// <summary>
-        /// Gets the 
+        /// Gets the correct count for the next read operation.
         /// </summary>
         /// <param name="count">The count requested to be read by the caller.</param>
         /// <returns>The remaining bytes to read within the range defined for this stream.</returns>
         private int PrepareStreamForRangeRead(int count)
         {
-            long effectiveCount = Math.Min(count, _totalCount - _currentCount);
-            if (effectiveCount > 0)
+            // A negative count causes base.Raad* methods to throw an ArgumentOutOfRangeException.
```

Comment:
Rewrote method because `_currentCount` may now exceed `_totalCount` and we need to distinguish that case (a no-op) from `count < 0` (an error).